### DISCUSSION TITLE
How to compile the python code to PYODIDE ?

### DISCUSSION THREAD

**Discussion by danjhou** [pyodide/pyodide#2441] [General]
I did a lot of codes for compiling my python code to pyodide, yet I encountered that the pyodide did not read the cv2 for calling open-cv.

[app.js.txt](https://github.com/pyodide/pyodide/files/8601193/app.js.txt)

**Comment by ryanking13** [comment]
`opencv.js` is not for Pyodide.

Try:

```js
await pyodide.loadPackage("opencv-python")
python.runPython("import cv2; print(cv2.__version__)")
```

**Reply by danjhou** [reply]
not working but thank you :)

**Reply by ryanking13** [reply]
Oh, there is a typo, sorry :/

```js
pyodide.runPython("import cv2; print(cv2.__version__)")
```

**Reply by danjhou** [reply]
App.js:27 Uncaught (in promise) TypeError: Failed to fetch dynamically imported module: https://cdn.jsdelivr.net/pyodide/dev/full/pyodide.js/pyodide.asm.js

are you familiar with this error? thank you.

**Reply by ryanking13** [reply]
Try removing indexURL parameter in loadPyodide. It is not required anymore.

**Reply by danjhou** [reply]
import { useEffect } from 'react';
import script from '.python/FaceMesh.py'
import './App.css';



const App = () => {

  useEffect(() => {
    const run = async () => {
      const scriptText = await (await fetch(script)).text();
      runScript(scriptText);
    }
    run();
  }, []);


  const runScript = async (code) => {
    const pyodide = await window.loadPyodide({
      indexURL : "https://cdn.jsdelivr.net/pyodide/dev/full/pyodide.js"
    });

    await pyodide.loadPackage("opencv-python")
    pyodide.runPython("import cv2; print(cv2.__version__)")

    await pyodide.runPythonAsync(code);
  }


  return (
    <div className="App">

    </div>
  );
}








can you show me? thank you.

**Reply by ryanking13** [reply]
From:

```suggestion
const pyodide = await window.loadPyodide({indexURL : "https://cdn.jsdelivr.net/pyodide/dev/full/pyodide.js"});
```

To:

```
const pyodide = await window.loadPyodide();
```

**Reply by danjhou** [reply]
pyodide.asm.js:10 Uncaught (in promise) PythonError: Traceback (most recent call last):
  File "/lib/python3.10/asyncio/futures.py", line 201, in result
    raise self._exception
  File "/lib/python3.10/asyncio/tasks.py", line 232, in __step
    result = coro.send(None)
  File "/lib/python3.10/site-packages/_pyodide/_base.py", line 500, in eval_code_async
    await CodeRunner(
  File "/lib/python3.10/site-packages/_pyodide/_base.py", line 353, in run_async
    await coroutine
  File "<exec>", line 1, in <module>
ModuleNotFoundError: No module named 'micropip'


Did you encounter this problem too? thank you

**Reply by ryanking13** [reply]
You should load `micropip` before using it.

```
await pyodide.loadPackage("micropip")
```

**Reply by danjhou** [reply]
are you familiar with this error too? thanks


opencv_python-4.4.0.40-cp38-cp38-win_amd64.whl' is not a pure Python 3 wheel

**Reply by ryanking13** [reply]
No, that's not normal. Can you share us your current code?

**Reply by danjhou** [reply]
import { useEffect } from 'react';
import script from './python/FaceMesh.py'
import './App.css';



const App = () => {

  useEffect(() => {
    const run = async () => {
      const scriptText = await (await fetch(script)).text();
      runScript(scriptText);
    }
    run();
  }, []);


  const runScript = async (code) => {
    const pyodide = await window.loadPyodide();
    await pyodide.loadPackage("micropip")
    await pyodide.loadPackage("opencv-python")
    pyodide.runPython("import cv2; print(cv2.__version__)")

    await pyodide.runPythonAsync(code);
  }


  return (
    <div className="App">

    </div>
  );
}

export default App;



I'm very confused regarding 3 wheel

**Reply by ryanking13** [reply]
Hmm that's very strange. 

> opencv_python-4.4.0.40-cp38-cp38-win_amd64.whl' is not a pure Python 3 wheel

This kind of error happens when you are trying to install non-pure Python wheel via `micropip.install()`. But `opencv-python` is loaded via `pyodide.loadPackage()` not `micropip.install()`. So I don't understand why that error happens.

Here is a minimal working example of loading opencv-python: https://jsfiddle.net/a7jwv951/8/.
